docs/examples/101/storage-static-website/main.bicep

```diff
@@ -0,0 +1,69 @@
+// This template deploys an Azure Storage account, and then configures it to support static website hosting.
+// Enabling static website hosting isn't possible directly in Bicep or an ARM template,
+// so this sample uses a deployment script to enable the feature.
+
+param location string
+param accountName string
+param skuName string {
+  allowed: [
+    'Standard_LRS'
+    'Standard_GRS'
+    'Standard_ZRS'
+    'Premium_LRS'
+  ]
+}
+param deploymentScriptTimestamp string = utcNow()
+param indexDocument string = 'index.html'
+param errorDocument404Path string = 'error.html'
+
+var storageAccountContributorRoleDefinitionId = subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '17d1049b-9a84-46fb-8f53-869881c3d3ab') // This is the Storage Account Contributor role, which is the minimum role permission we can give. See https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#:~:text=17d1049b-9a84-46fb-8f53-869881c3d3ab
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2019-06-01' = {
+  name: accountName
+  location: location
+  kind: 'StorageV2'
+  sku: {
+    name: skuName
+  }
+  properties: {}
+}
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: 'DeploymentScript'
+  location: location
+}
+
+resource roleAssignment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+  scope: storageAccount
+  name: guid(resourceGroup().id)

```

Comment:
I would add the `storageAccountContributorRoleDefinitionId` to the guid() function. With just the resourceGroup ID you will be more likely to get roleAssignment name conflicts.

Reply:
Thanks - fixed.